src/benchmarking.rs

```diff
@@ -17,12 +17,672 @@
+
+type RuntimeOrigin<T> = <T as frame_system::Config>::RuntimeOrigin;
+const TRB: u128 = 10u128.pow(DECIMALS);
+const SEED: u32 = 0;
+
+fn assert_last_event<T: Config>(generic_event: <T as Config>::RuntimeEvent) {
+	frame_system::Pallet::<T>::assert_last_event(generic_event.into());
+}
+
+fn trb(amount: impl Into<f64>) -> Tributes {
+	// TRB amount has 18 decimals
+	Tributes::from((amount.into() * TRB as f64) as u128)
+}
+
+fn token<T: Config>(amount: impl Into<u64>) -> BalanceOf<T> {
+	// test parachain token
+	(amount.into() * unit::<T>() as u64).into()
+}
+
+fn unit<T: Config>() -> u128 {
+	let decimals: u8 = T::Decimals::get();
+	10u128.pow(decimals.into())
+}
+
+fn uint_value<T: Config>(value: impl Into<Uint>) -> ValueOf<T> {
+	ethabi::encode(&[Token::Uint(value.into())]).try_into().unwrap()
+}
+
+fn spot_price(asset: impl Into<String>, currency: impl Into<String>) -> Bytes {
+	ethabi::encode(&[
+		Token::String("SpotPrice".to_string()),
+		Token::Bytes(ethabi::encode(&[
+			Token::String(asset.into()),
+			Token::String(currency.into()),
+		])),
+	])
+}
+
+fn deposit_stake<T: Config>(
+	reporter: AccountIdOf<T>,
+	amount: Tributes,
+	address: Address,
+) -> Result<RuntimeOrigin<T>, BenchmarkError> {
+	match T::StakingOrigin::try_successful_origin() {
+		Ok(origin) => {
+			Tellor::<T>::report_stake_deposited(origin.clone(), reporter, amount, address)
+				.map_err(|_| BenchmarkError::Weightless)?;
+			Ok(origin)
+		},
+		Err(_) => Err(BenchmarkError::Weightless),
+	}
+}
+
+// Helper function for creating feeds
+fn create_feed<T: Config>(
+	feed_creator: AccountIdOf<T>,
+	query_id: QueryId,
+	reward: BalanceOf<T>,
+	start_time: Timestamp,
+	interval: Timestamp,
+	window: Timestamp,
+	price_threshold: u16,
+	reward_increase_per_second: BalanceOf<T>,
+	query_data: QueryDataOf<T>,
+	amount: BalanceOf<T>,
+) -> FeedId {
+	Tellor::<T>::setup_data_feed(
+		RawOrigin::Signed(feed_creator).into(),
+		query_id,
+		reward,
+		start_time,
+		interval,
+		window,
+		price_threshold,
+		reward_increase_per_second,
+		query_data.clone(),
+		amount,
+	)
+	.unwrap();
+	let feed_id = Keccak256::hash(&ethabi::encode(&vec![
+		Token::FixedBytes(query_id.0.into()),
+		Token::Uint(reward.into()),
+		Token::Uint(start_time.into()),
+		Token::Uint(interval.into()),
+		Token::Uint(window.into()),
+		Token::Uint(price_threshold.into()),
+		Token::Uint(reward_increase_per_second.into()),
+	]))
+	.into();
+	feed_id
+}
+
+fn dispute_id(para_id: u32, query_id: QueryId, timestamp: Timestamp) -> DisputeId {
+	Keccak256::hash(&ethabi::encode(&[
+		Token::Uint(para_id.into()),
+		Token::FixedBytes(query_id.0.to_vec()),
+		Token::Uint(timestamp.into()),
+	]))
+	.into()
+}
 
 benchmarks! {
-	impl_benchmark_test_suite!(Template, crate::mock::new_test_ext(), crate::mock::Test);
+
+
+	register {
+
+	}: _(RawOrigin::Root)
+	verify {
+		assert_last_event::<T>(
+				Event::RegistrationSent { para_id: 2_000, contract_address: T::Registry::get().address.into() }.into(),
```

Comment:
Will this not fail with para_id being hardcoded to 2000, depending on runtime config? Can it not get it from the para_id of the Registry contract location in config?

Like done with staking contract elsewhere in benchmarks:
let staking_contract = T::Staking::get();

Reply:
I've removed this as we are removing it in upcoming PR